src/fileActions.jsx

```diff
@@ -209,3 +216,90 @@ export const RenameItemModal = ({ path, setPath, selected }) => {
+    const [ownerAccess, setOwnerAccess] = useState(selected.permissions.slice(0, 1));
+    const [group, setGroup] = useState(selected.group);
+    const [groupAccess, setGroupAccess] = useState(selected.permissions.slice(1, 2));
+    const [otherAccess, setOtherAccess] = useState(selected.permissions.slice(2, 3));
+    const [errorMessage, setErrorMessage] = useState(undefined);
+
+    const editPermissions = () => {
+        const options = { err: "message", superuser: "try" };
+        const command = ["chmod", ownerAccess + groupAccess + otherAccess, "/" + path.join("/") + "/" + selected.name];
+
+        cockpit.spawn(command, options)
+                .then(() =>
+                    cockpit.spawn(["chown", owner + ":" + group, "/" + path.join("/") + "/" + selected.name], options)
+                )
+                .then(Dialogs.close, err => { setErrorMessage(err.message) });
+    };
+
+    const permissions = [
+        { label: _("None"), value: "0" },
+        { label: _("Read-only"), value: "4" },
+        { label: _("Write-only"), value: "2" },
+        { label: _("Execute-only"), value: "1" },
+        { label: _("Read and write"), value: "6" },
+        { label: _("Read and execute"), value: "5" },
+        { label: _("Read, write and execute"), value: "7" },
+        { label: _("Write and execute"), value: "3" },
+    ];
+
+    return (
+        <Modal
+          position="top"
+          title={_("Edit permissions")}
+          isOpen
+          onClose={Dialogs.close}
+          footer={
+              <>
+                  <Button variant="primary" onClick={editPermissions}>{_("Change")}</Button>
+                  <Button variant="link" onClick={Dialogs.close}>{_("Cancel")}</Button>
+              </>
+          }
+        >
+            <Stack>
+                {errorMessage !== undefined &&
+                <InlineNotification
+                  type="danger"
+                  text={errorMessage}
+                  isInline
+                />}
+                <Form isHorizontal>
+                    <FormGroup label={_("Owner")} fieldId="edit-permissions-owner">
+                        <TextInput value={owner} onChange={setOwner} id="edit-permissions-owner" />
+                    </FormGroup>
+                    <FormGroup label={_("Owner access")} fieldId="edit-permissions-owner-access">
+                        <FormSelect value={ownerAccess} onChange={(_, val) => { setOwnerAccess(val) }} id="edit-permissions-owner-access">
+                            {permissions.map(p => (
+                                <FormSelectOption key={p.value} value={p.value} label={p.label} />
+                            ))}
+                        </FormSelect>
+                    </FormGroup>
+                    <Divider />
+                    <FormGroup label={_("Group")} fieldId="edit-permissions-group">
+                        <TextInput value={group} onChange={setGroup} id="edit-permissions-group" />
```

Comment:
So I was wondering - if it's more correct for the group selection to be a select- which takes the contents of /etc/groups filtered with the non-system groups. What do you think? Maybe followup? @garrett?

Reply:
I agree, let's hear Garrett's opinion too